Demo/Demo/PayPalWebPayments/PayPalWebViewModel/PayPalWebViewModel.swift

```diff
@@ -44,51 +50,64 @@ class PayPalWebViewModel: ObservableObject {
         } catch {
-            updateState(.error(message: error.localizedDescription))
+            DispatchQueue.main.async {
+                self.state.createdOrderResponse = .error(message: error.localizedDescription)
+            }
             print("❌ failed to fetch orderID with error: \(error.localizedDescription)")
         }
     }
 
     func paymentButtonTapped(funding: PayPalWebCheckoutFundingSource) {
         Task {
             do {
-                self.updateState(.loading)
+                DispatchQueue.main.async {
+                    self.state.approveResultResponse = .loading
+                }
                 payPalWebCheckoutClient = try await getPayPalClient()
                 guard let payPalWebCheckoutClient else {
                     print("Error initializing PayPalWebCheckoutClient")
                     return
                 }
 
-                if let orderID {
+                if let orderID = state.createOrder?.id {
                     let payPalRequest = PayPalWebCheckoutRequest(orderID: orderID, fundingSource: funding)
                     payPalWebCheckoutClient.start(request: payPalRequest) { result, error in
                         if let error {
-                            if error == PayPalError.checkoutCanceledError {
-                                print("Canceled")
-                                self.updateState(.idle)
-                            } else {
-                                self.updateState(.error(message: error.localizedDescription))
+                            DispatchQueue.main.async {
+                                if error == PayPalError.checkoutCanceledError {
+                                    print("Canceled")
+                                    self.state.approveResultResponse = .idle
+                                } else {
+                                    self.state.approveResultResponse = .error(message: error.localizedDescription)
+                                }
                             }
                         } else {
-                            self.updateState(.success)
-                            self.checkoutResult = result
+                            DispatchQueue.main.async {
+                                self.state.approveResultResponse = .loaded(
+                                    PayPalPaymentState.ApprovalResult(id: orderID, status: "APPROVED")
+                                )
+                                self.checkoutResult = result
+                                print("✅ Checkout result: \(String(describing: result))")
+                            }
                         }
                     }
                 }
-                updateState(.success)
             } catch {
                 print("Error starting PayPalWebCheckoutClient")
-                updateState(.error(message: error.localizedDescription))
+                self.state.createdOrderResponse = .error(message: error.localizedDescription)
```

Comment:
I think this needs to be on main also

Reply:
Just a little fix here and it looks really good Georgi. Great work!
Thanks you!